onnxruntime/core/session/inference_session.cc

```diff
@@ -87,6 +160,38 @@ inline std::basic_string<T> GetCurrentTimeString() {
   return std::basic_string<T>(time_str);
 }
 }  // namespace
+struct CustomOpKernel : OpKernel {
+  CustomOpKernel(const OpKernelInfo& info, OrtCustomOp& op) : OpKernel(info), op_(op) {
+    op_.CreateKernel(&op_, reinterpret_cast<OrtKernelInfo*>(const_cast<OpKernelInfo*>(&info)), &op_kernel_);
+  }
+
+  ~CustomOpKernel() {
+    op_.KernelDestroy(op_kernel_);
+  }
+
+  Status Compute(OpKernelContext* ctx) const override {
```

Comment:
>OpKernelContext [](start = 17, length = 15)

We control running these kernels right? In that case, we actually have an OpKernelContextInternal (control flow nodes require that), so the MLValue could have been accessed via casting ctx to that type. It was added that way as I was told we wanted to keep OpKernelContext as clean as possible.